examples/main.go

```diff
@@ -23,15 +23,269 @@ func main() {
 		log.Fatalf("did not connect: %v", err)
 	}
 	defer conn.Close()
-	c := pb.NewQdrantClient(conn)
+
+	// create grpc collection client
+	collections_client := pb.NewCollectionsClient(conn)
 
 	// Contact the server and print out its response.
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
-	r, err := c.HealthCheck(ctx, &pb.HealthCheckRequest{})
+
+	// Check Qdrant version
+	qdrantClient := pb.NewQdrantClient(conn)
+	healthCheckResult, err := qdrantClient.HealthCheck(ctx, &pb.HealthCheckRequest{})
+	if err != nil {
+		log.Fatalf("Could not get health: %v", err)
+	} else {
+		log.Printf("Qdrant version: %s", healthCheckResult.GetVersion())
+	}
+
+	collectionName := "test_collection"
+
+	// Delete collection
+	_, err = collections_client.Delete(ctx, &pb.DeleteCollection{
+		CollectionName: collectionName,
+	})
+	if err != nil {
+		log.Fatalf("Could not delete collection: %v", err)
+	} else {
+		log.Println("Collection", collectionName, "deleted")
+	}
+
+	// Create new collection
+	var defaultSegmentNumber uint64 = 2
+	_, err = collections_client.Create(ctx, &pb.CreateCollection{
+		CollectionName: collectionName,
+		VectorSize:     4,
+		Distance:       pb.Distance_Dot,
+		OptimizersConfig: &pb.OptimizersConfigDiff{
+			DefaultSegmentNumber: &defaultSegmentNumber,
+		},
+	})
+	if err != nil {
+		log.Fatalf("Could not create collection: %v", err)
+	} else {
+		log.Println("Collection", collectionName, "created")
+	}
+
+	// List all created collections
+	r, err := collections_client.List(ctx, &pb.ListCollectionsRequest{})
+	if err != nil {
+		log.Fatalf("Could not get collections: %v", err)
+	} else {
+		log.Printf("List of collections: %s", r.GetCollections())
+	}
+
+	// Create points grpc client
+	pointsClient := pb.NewPointsClient(conn)
+
+	// Create keyword field index
+	fieldIndex1Type := pb.FieldType_FieldTypeKeyword
+	fieldIndex1Name := "city"
+	_, err = pointsClient.CreateFieldIndex(ctx, &pb.CreateFieldIndexCollection{
+		CollectionName: collectionName,
+		FieldName:      fieldIndex1Name,
+		FieldType:      &fieldIndex1Type,
+	})
+	if err != nil {
+		log.Fatalf("Could not create field index: %v", err)
+	} else {
+		log.Println("Field index for field", fieldIndex1Name, "created")
+	}
+
+	// Create integer field index
+	fieldIndex2Type := pb.FieldType_FieldTypeInteger
+	fieldIndex2Name := "count"
+	_, err = pointsClient.CreateFieldIndex(ctx, &pb.CreateFieldIndexCollection{
+		CollectionName: collectionName,
+		FieldName:      fieldIndex2Name,
+		FieldType:      &fieldIndex2Type,
+	})
+	if err != nil {
+		log.Fatalf("Could not create field index: %v", err)
+	} else {
+		log.Println("Field index for field", fieldIndex2Name, "created")
+	}
+
+	// Upsert points
+	waitUpsert := true
+	upsertPoints := []*pb.PointStruct{
```

Comment:
This long array makes it hard to read the example. maybe moving it to somewhere like `mock_data.go`?

Reply:
Agree but I would keep in this file because I want to merge these changes asap. Maybe as next PR